src/Pipeline.cpp

```diff
@@ -665,7 +674,7 @@ Callable Pipeline::compile_to_callable(const std::vector<Argument> &args_in, con
 
     for (const auto &out : outputs) {
         for (Type t : out.output_types()) {
-            args.emplace_back(out.name(), Argument::OutputBuffer, t, out.dimensions(), ArgumentEstimates{});
+            args.emplace_back(strip_uniquifier(out.name()), Argument::OutputBuffer, t, out.dimensions(), ArgumentEstimates{});
```

Comment:
What does this stripped name get used for? It's currently legal to write nonsense like:

```
#include "Halide.h"

using namespace Halide;

int main(int argc, char **argv) {
    Func f1("f"), f2("f");
    Var x;
    f1(x) = x;
    f2(x) = x + 1;

    Pipeline p({f1, f2});

    p.compile_jit();

    return 0;
}
```

but that still seems to work in this branch

Reply:
OK, revisiting this: the issue here is that we are building up a list of Arguments to use for the Callable; for most uses, the name is just "decoration" (e.g. for annotating the function prototype in the header). In this case, though, we're going to use them to populate the list of `kwargs` in the Python wrappers, so that we can do something like `simple(buffer_input=b_in, float_arg=float_in, simple_output=b_out)`. 

The gotcha, of course, is that Outputs in a Generator are essentially Funcs, and the names get uniquified as a result, so if you the same Generator multiple times (or different Generators with matching output names), you'll get outputs with names like `simple_output$4`, which are unusable as kwargs.

This change was really just a quick hack, on the assumption that any Argument we see that ends in `$NUM` is just a uniquified name and that we can trim off that bit to get the 'real' name.

I can revisit the code paths and see if there is way to ensure that the 'original' Output names get transmitted from the Generator all the way down this path, but it would likely take the addition of some new data channel to make that happen reliably, which is why I took the quick way out.

Given the use case here, I suspect that just doing the stripping is safe, with the caveat that you *could* construct a pipeline like this that would be ambiguous to call via kwargs in Python -- but not via a Generator, since they enforce that all inputs and outputs have unique names -- so it would be a case of "don't do that".

Reply:
(Gentle ping to @abadams for feedback on my feedback)

Reply:
I'm mostly concerned about the non-python non-Generator case, because as you say name collisions are checked in a Generator. I'm worried that we'll have two sets of out-of-sync names floating around that produce hard-to-debug errors later due to the mismatch if someone does something creative with a Callable, e.g. by using names from Callable::arguments() to go look up into a map of Funcs by name. 

Maybe it would be safer to strip the $* at the point where the kwargs are populated.

Reply:
Good call. f613271835688244830b68379de2543fb9bfc59b has the fix, I think. PTAL.